components/vocabularies/api/src/main/java/org/phenotips/vocabulary/VocabularyManager.java

```diff
@@ -44,6 +44,16 @@
     VocabularyTerm resolveTerm(String termId);
 
     /**
+     * Retrieve a term given its owner vocabulary. The owner vocabulary's id is specified.
+     * @param vocabularyId the identifier for the owner vocabulary, for example {@code HGNC}
+     * @param termId the term identifier, in the format {@code <vocabulary prefix>:<term id>}, for example
+     *               {@code HP:0002066}
+     * @return the requested term, or {@code null} if the term doesn't exist in the vocabulary, or no matching
+     *         vocabulary is available
+     */
+    VocabularyTerm resolveTerm(String vocabularyId, String termId);
```

Comment:
I'd rather not add a new method here, for API stability. It's easy to just use this instead:

    final VocabularyTerm term = vocabularies.getVocabulary(HGNC).getTerm(geneSymbol);

Or, instead of working with the vocabulary manager, you can just do:

    @Inject @Named('hgnc') private Vocabulary hgnc;
    ....
    final VocabularyTerm term = hgnc.getTerm(geneSymbol);